pkg/sink/kafka/v2/kafka-go-client.go

```diff
@@ -0,0 +1,363 @@
+	return metrics.DefaultRegistry
+}
+
+// Close closes the client
+func (k *kafkaGoClient) Close() error {
+	return nil
+}
+
+type syncWriter struct {
+	w *kafka.Writer
+}
+
+func (s *syncWriter) SendMessage(topic string,
+	partitionNum int32, key []byte, value []byte,
+) error {
+	return s.w.WriteMessages(context.Background(), kafka.Message{
+		Topic:     topic,
+		Partition: int(partitionNum),
+		Key:       key,
+		Value:     value,
+	})
+}
+
+// SendMessages produces a given set of messages, and returns only when all
+// messages in the set have either succeeded or failed. Note that messages
+// can succeed and fail individually; if some succeed and some fail,
+// SendMessages will return an error.
+func (s *syncWriter) SendMessages(topic string,
+	partitionNum int32,
+	key []byte, value []byte,
+) error {
+	msgs := make([]kafka.Message, int(partitionNum))
+	for i := 0; i < int(partitionNum); i++ {
+		msgs[i] = kafka.Message{
+			Topic:     topic,
+			Key:       key,
+			Value:     value,
+			Partition: i,
+		}
+	}
+	return s.w.WriteMessages(context.Background(), msgs...)
+}
+
+// Close shuts down the producer; you must call this function before a producer
+// object passes out of scope, as it may otherwise leak memory.
+// You must call this before calling Close on the underlying client.
+func (s *syncWriter) Close() error {
+	return s.w.Close()
+}
+
+type asyncWriter struct {
+	w            *kafka.Writer
+	changefeedID model.ChangeFeedID
+	closedChan   chan struct{}
+	failpointCh  chan error
+	successes    chan []kafka.Message
+	errorsChan   chan error
+}
+
+// AsyncClose triggers a shutdown of the producer. The shutdown has completed
+// when both the Errors and Successes channels have been closed. When calling
+// AsyncClose, you *must* continue to read from those channels in order to
+// drain the results of any messages in flight.
+func (a *asyncWriter) AsyncClose() {
+	go func() {
+		_ = a.Close()
+	}()
+}
+
+// Close shuts down the producer and waits for any buffered messages to be
+// flushed. You must call this function before a producer object passes out of
+// scope, as it may otherwise leak memory. You must call this before process
+// shutting down, or you may lose messages. You must call this before calling
+// Close on the underlying client.
+func (a *asyncWriter) Close() error {
+	return a.w.Close()
+}
+
+// AsyncSend is the input channel for the user to write messages to that they
+// wish to send.
+func (a *asyncWriter) AsyncSend(ctx context.Context, topic string,
+	partition int32, key []byte, value []byte,
+	callback func(),
+) error {
+	select {
+	case <-ctx.Done():
+		return errors.Trace(ctx.Err())
+	case <-a.closedChan:
+		log.Warn("Receive from closed chan in kafka producer",
+			zap.String("namespace", a.changefeedID.Namespace),
+			zap.String("changefeed", a.changefeedID.ID))
+		return nil
+	default:
+	}
+	return a.w.WriteMessages(context.Background(), kafka.Message{
```

Comment:
can we used the passed in ctx directly, instead of the `background`.

Reply:
I will refine the interface and add ctx